### DISCUSSION TITLE
How do I create custom utilities and import them into a controller

### DISCUSSION THREAD

**Discussion by simonjcarr** [adonisjs/core#4261] [Help]
I want to create a custom utility to work with keycloak.

I created the following folder structure in my adonis project

```
app
----keycloak
     ----token.ts
```

Inside token.ts I have the following code 

```js
import axios from 'axios'
import Env from '@ioc:Adonis/Core/Env'
export async function getKeycloakAdminToken() {
  let tokenResponse
    try {
    tokenResponse = await axios.post(`${Env.get('KC_URL')}/auth/realms/${Env.get('KC_REALM')}/protocol/openid-connect/token`, {
      grant_type: 'password',
      client_id: Env.get('KC_CLIENT_ID'),
      username: Env.get('KC_ADMIN_USERNAME'),
      password: Env.get('KC_ADMIN_PASSWORD'),
      client_secret: Env.get('KC_CLIENT_SECRET'),
    }, {
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded',
      },
    })
  } catch (error) {
    console.log(error)
  }

    const { access_token } = tokenResponse.data
    return access_token
}

```

In my controller I have tried the following two imports but in both cases I get a cant find module error

```
import { getKeycloakAdminToken } from '../../keycloak/token'
import { getKeycloakAdminToken } from 'App/keycloak/token'
```
Cannot find module '../../keycloak/token' or its corresponding type declarations.
Cannot find module 'App/keycloak/token' or its corresponding type declarations.

I assume it is because typescript is not compiling the file at build time, but how do I add custom utilities in Adonis then?

**Comment by thetutlage** [comment]
What happens when you run `npm run build`? Please share the error log of this command
